adafruit_max31856.py

```diff
@@ -0,0 +1,291 @@
+_MAX31856_CJTL_REG          = const(0x0B)
+_MAX31856_LTCBH_REG         = const(0x0C)
+_MAX31856_LTCBM_REG         = const(0x0D)
+_MAX31856_LTCBL_REG         = const(0x0E)
+_MAX31856_SR_REG            = const(0x0F)
+
+# fault types
+_MAX31856_FAULT_CJRANGE     = const(0x80)
+_MAX31856_FAULT_TCRANGE     = const(0x40)
+_MAX31856_FAULT_CJHIGH      = const(0x20)
+_MAX31856_FAULT_CJLOW       = const(0x10)
+_MAX31856_FAULT_TCHIGH      = const(0x08)
+_MAX31856_FAULT_TCLOW       = const(0x04)
+_MAX31856_FAULT_OVUV        = const(0x02)
+_MAX31856_FAULT_OPEN        = const(0x01)
+
+
+  
+# an enum-like class for the thermocouple types and their representations
+class ThermocoupleType:
+  B  = 0b0000
+  E  = 0b0001
+  J  = 0b0010
+  K  = 0b0011
+  N  = 0b0100
+  R  = 0b0101
+  S  = 0b0110
+  T  = 0b0111
+  G8  = 0b1000
+  G32 = 0b1100
+
+class MAX31856:
+  """Driver for the MAX31856 Universal Thermocouple Amplifier
+  
+    :param ~busio.I2C i2c_bus: The I2C bus the DS3231 is connected to.
+    :param int address: The I2C address of the device.
+  """
+
+  # A class level buffer to reduce allocations for reading and writing. 
+  # Tony says this isn't re-entrant or thread safe!
+  _BUFFER = bytearray(3)
+
+  def __init__(self, spi, cs, thermocouple_type=ThermocoupleType.K):
+    self._device = SPIDevice(spi, cs, baudrate=500000,
+                                            polarity=0, phase=1)
+    
+    # # assert on any fault
+    self._write_u8(_MAX31856_MASK_REG, 0x0)
+    # turn off open circuit faults?
+    self._write_u8(_MAX31856_CR0_REG, _MAX31856_CR0_OCFAULT0)
+    
+    # set thermocouple type
+    # get current value of CR1 Reg
+    t = self._read_u8(_MAX31856_CR1_REG)
+    t &= 0xF0 #mask off bottom 4 bits
+    # add the new value for the TC type
+    t |= int(thermocouple_type) & 0x0F
+    self._write_u8(_MAX31856_CR1_REG, t)
+
+  @property 
+  def temperature(self):
+    """Read the temperature of the sensor and return its value in degrees
+    Celsius. (read-only)
+    """
+    self._perform_one_shot_measurement()
+
+    # read three bytes starting at the MSB of the 
+    raw_read = self._read_u24(_MAX31856_LTCBH_REG)
+    
+    # fix sign if needed
+    if (raw_read & 0x800000):
+      raw_read |= 0xFF000000 
+    
+    # bottom 5 bits are unused
+    raw_read >>= 5 
+    # scale back from linearized value to temp
+    temp_float = raw_read * 0.0078125
+
+    return temp_float
+
+  @property
+  def reference_temperature(self):
+    """Read the temperature of the cold junction and return its value in degrees
+    Celsius. (read-only)
+    """
+    self._perform_one_shot_measurement()
+    raw_read = self._read_u16(_MAX31856_CJTH_REG)
+    cold_junction_temp = raw_read / 256.0
+
+    return cold_junction_temp
+
+  @temperature_thresholds.setter
+  def temperature_thresholds(self, val):
+    """Set the thermocouple's low and high temperature thresholds as a tuple: `thermocouple.temperature_threshold = (low_threshold, high_threshold)`."""
```

Comment:
You only need the getter documented for ReadTheDocs. The property comments should read like its a variable and not a function (So no "Read" or "Set" to begin with.)

Reply:
I specifically documented the setter like this as it wasn't obvious to me that you would be able to assign it a tuple. Perhaps because the getter returns a tuple it should be obvious? Any specific recommendations would be welcome. 

I take your second comment to mean the getter documentation should be:
>The thermocouple's low and high temperature thresholds as a `(low_temp, high_temp)` tuple

Now that I write that it does seem to imply that you would set it with a tuple as well..

Reply:
Yup, exactly. You can document the type of the "variable"